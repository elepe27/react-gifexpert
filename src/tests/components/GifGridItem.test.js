import { shallow } from 'enzyme/build';
import {GifGridItem} from '../../components/GifGridItem'


describe('Pruebas en GifGridItem', () => {
   
    const title = "Imagen de goku";
    const url = "https://localhost/algo.jpg"
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
    
    test('debe de mostrar el componente correctamente', () => {


        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe de tener un parrafo con el titulo', () => {
    
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
        
        
    });

    test('debe tener la imagen igual al url y alt de los props', () => {
       
        
        const img = wrapper.find('img');
        // console.log( img.prop('src') );
        
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('debe tener animate__fadeIn', () => {
       const div = wrapper.find('div');
       expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
    //    expect( div.prop('class').contain() )
        
    });
    
});