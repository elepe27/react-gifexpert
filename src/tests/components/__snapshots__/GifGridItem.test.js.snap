// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Imagen de goku"
    src="https://localhost/algo.jpg"
  />
  <p>
    Imagen de goku
  </p>
</div>
`;
